### PR TITLE
added remove all on select

### PR DESCRIPTION
**Pull request by priyankafnu**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added
- [ ] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Style Update (CSS)
- [ ] Other... Please describe:

## What is the current behavior?
When a lot of results are selected, the "clear all" button appears way down the page in an unusual position.

## What is the new behavior?
This adds a "remove all" button at the top of the select dropdown when items are selected to allow a user to easily remove all options when lots are selected.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-select/go-select.component.html [hunk 1]
```diff
@@ -39,6 +39,12 @@
         [ngClass]="{ 'go-select__select-all-button--dark' : theme === 'dark' }">
         Select All
       </button>
+      <button
+      (click)="onRemoveAll()"
+      class="go-select__select-all-button"
+      [ngClass]="{ 'go-select__select-all-button--dark' : theme === 'dark' }">
+      Remove  All
+    </button>
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.html:47]: Should we have a conditional here to only render the button if the value of the control is set? Not sure it makes sense from a UX perspective to show a "Remove All" option if they haven't selected anything.

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts [hunk 2]
```diff
@@ -56,6 +56,10 @@ export class GoSelectComponent implements OnInit {
     this.control.patchValue(this.items.map((item: any) => this.bindValue ? item[this.bindValue] : item));
   }
 
+  onRemoveAll(): void {
+    this.control.reset();
+  }
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.ts:61]: Should we write a test for this method?

**Review by grahamhency — CHANGES_REQUESTED** [review]
Functionality looks good. I had one additional suggestion and a comment on tests.

File: projects/go-lib/src/lib/components/go-select/go-select.component.html (hunk 1 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html:51]: @grahamhency I have added showRemoveAll condition. Please check and let me know if it is okay or not

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts (hunk 2 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.ts:62]: @grahamhency test case done. Please check

File: projects/go-lib/src/lib/components/go-select/go-select.component.html (hunk 1 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html:51]: @priyankafnu my question still stands. The change that you've put up to add the `showRemoveAll` binding here will always be true unless someone passes in the value to `go-select` of `true`. I don't think we need the `showRemoveAll` binding here. We can probably just do:
```
<ng-container *ngIf="multiple && items?.length">

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts [hunk 3]
```diff
@@ -35,6 +35,7 @@ export class GoSelectComponent implements OnInit {
   @Input() placeholder: string;
   @Input() searchable: boolean = true;
   @Input() showSelectAll: boolean = true;
+  @Input() showRemoveAll: boolean = false;
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.ts:38]: As stated in my other comment, I don't believe we need this. If you wanted to keep it to be able to override showing/hiding the button, then this should default to `true` and we can keep the binding in the conditional.

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts [hunk 4]
```diff
@@ -102,4 +102,19 @@ describe('GoSelectComponent', () => {
       expect(component.control.value).toEqual([1, 2, 3]);
     });
   });
+
+  describe('onRemoveAll()', () => {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts:106]: This should just be `onRemoveAll`, the parenthesis are not needed.

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts [hunk 5]
```diff
@@ -102,4 +102,19 @@ describe('GoSelectComponent', () => {
...
+    it('uses bindValue to get value if bindValue exists', () => {
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts:107]: We should change the description here to be something related to what we're testing.

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts [hunk 6]
```diff
@@ -102,4 +102,19 @@ describe('GoSelectComponent', () => {
...
+      component.bindValue = 'id';
+      component.items = [
+        { id: 1, label: 'Label 1' },
+        { id: 2, label: 'Label 2' },
+        { id: 3, label: 'Label 3' }
+      ];
+
+      component.onRemoveAll();
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts:115]: Should we set `component.control` to a value prior calling this method to ensure that the control actually goes from a value to null?

**Review by grahamhency — CHANGES_REQUESTED** [review]

File: projects/go-lib/src/lib/components/go-select/go-select.component.html (hunk 1 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html:51]: Also, we really should only show this button if the control has a value. So the final conditional should look like:
```
<ng-container *ngIf="multiple && items?.length && control.value">

File: projects/go-lib/src/lib/components/go-select/go-select.component.ts (hunk 3 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.ts]: Hi @grahamhency Done with the changes. Please check

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts (hunk 4 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts]: Hi @grahamhency 
Removed parenthesis from onRemoveAll. Please check

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts (hunk 6 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts:117]: Done this changes

File: projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts (hunk 5 above)
Comment (priyankafnu) [reply projects/go-lib/src/lib/components/go-select/go-select.component.spec.ts]: Done the changes

File: projects/go-lib/src/lib/components/go-select/go-select.component.html (hunk 1 above)
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html:51]: @priyankafnu, I still think that we should only render this button when the control has a value. Would you mind adding it to the conditional?

```
<ng-container *ngIf="multiple && items?.length && control.value">

**Review by grahamhency — CHANGES_REQUESTED** [review]
Just one more comment, but if you can address that I'll approve and we can get a new version ready.

**Review by grahamhency — APPROVED** [review]

**Pull request merged**